control-plane/typeSpecConverter.cpp

```diff
@@ -111,6 +115,50 @@ bool TypeSpecConverter::preorder(const IR::Type_Name* type) {
     return false;
 }
 
+// This function is invoked if an architecure's model .p4 file has a Newtype.
+// The function is not invoked for user-defined NewType.
+bool TypeSpecConverter::preorder(const IR::Type_Newtype* type) {
+    if (p4RtTypeInfo) {
+        bool orig_type = true;
+        const IR::StringLiteral* uri;
+        const IR::Constant* sdnB;
+        auto ann = type->getAnnotation("p4runtime_translation");
+        if (ann != nullptr) {
+            orig_type = false;
+            uri = ann->expr[0]->to<IR::StringLiteral>();
+            if (!uri)
+                ::error("P4runtime annotation does not have uri: %1%",
```

Comment:
This code is not written very well IMO. In case of an error here, `uri` will be set to `nullptr` and the code will segfault later on when you do `dataType->set_uri(std::string(uri->value));`. Same for `sdnB`.

Reply:
Sorry, I thought `::error` would exit the running program.  I will change the code.